### PR TITLE
ES-1992: migrate to corda runtime plugin

### PR DESCRIPTION
**Pull request by anton-subbotin**
[https://r3-cev.atlassian.net/browse/ES-1992](https://r3-cev.atlassian.net/browse/ES-1992)

**Not for merge!**

The goal is to migrate this code to a different repository; PR raised only to allow reviews of the changes.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -6,40 +8,44 @@ plugins {
...
+    cordaRuntimeGradlePlugin {
         cordaClusterURL = "https://localhost:8888"
-        networkConfigFile = "config/static-network-config.json"
-        r3RootCertFile = "config/r3-ca-key.pem"
+        cordaRestUser = "admin"
+        cordaRestPasswd ="admin"
+        cordaRuntimePluginWorkspaceDir = "workspace"
+        composeFilePath = "config/combined-worker-kafka-compose.yaml"
+        // Alternatively, you can use a Database-only combined worker:
+        // composeFilePath = "config/combined-worker-compose.yaml"
```
Comment (anton-subbotin) [inline build.gradle]: Do we need to provide an example of no-kafka deployment?

File: gradle.properties [hunk 2]
```diff
@@ -2,21 +2,17 @@ kotlin.code.style=official
...
+cordaNotaryPluginsVersion=5.2.0.0-RC02
 
 # Specify the version of the cordapp-cpb and cordapp-cpk plugins
-cordaPluginsVersion=7.0.3
+cordaPluginsVersion=7.0.4
 
-# Specify the version of the CSDE gradle plugin to use
-csdePluginVersion=1.2.0-beta-+
+# Specify the version of the Corda runtime Gradle plugin to use
+cordaGradlePluginVersion=5.2.0.0-RC02
```
Comment (anton-subbotin) [inline gradle.properties:15]: This and other versions will need to be updated after 5.2 release

File: config/combined-worker-kafka-compose.yaml [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    ]
+
+  corda:
+    image: corda-os-docker.software.r3.com/corda-os-combined-worker-kafka:5.2.0.0-RC02
+    depends_on:
+      - postgresql
+      - kafka
+      - kafka-create-topics
+    environment:
+      JAVA_TOOL_OPTIONS: -agentlib:jdwp=transport=dt_socket,server=y,suspend=n,address=*:5005
```
Comment (anton-subbotin) [inline config/combined-worker-kafka-compose.yaml]: This piece is missing in the examples provided in the [plugin's readme](https://github.com/corda/corda-runtime-os/tree/release/os/5.2/tools/corda-runtime-gradle-plugin#readme).
This is necessary to enable debugger connection on port 5005 - to provide the string in the JAVA_TOOL_OPTIONS variable.
For some reason, if we specify it as one of the args in the `command` list, it doesn't seem to be applied by JVM (as it does when started on a local JVM outside container), and debug connection fails.

File: README.md [hunk 4]
```diff
@@ -1,25 +1,22 @@
-# CSDE-cordapp-template-kotlin
+# release-V5 cordapp-template-kotlin
```
Comment (anton-subbotin) [inline README.md:1]: previously this was the name of the repository; now it is a specific branch name where the code would live, and the target repo name

File: contracts/src/main/kotlin/com/r3/developers/cordapptemplate/utxoexample/contracts/ChatContract.kt [hunk 5]
```diff
@@ -1,6 +1,6 @@
-package com.r3.developers.csdetemplate.utxoexample.contracts
+package com.r3.developers.cordapptemplate.utxoexample.contracts
```
Comment (anton-subbotin) [inline contracts/src/main/kotlin/com/r3/developers/cordapptemplate/utxoexample/contracts/ChatContract.kt:1]: rename package - change `csdetemplate` to `cordapptemplate`; I'm open for suggestions of a better name

File: build.gradle [hunk 6]
```diff
@@ -6,40 +8,45 @@ plugins {
...
-        cordaProcessorTimeout = processorTimeout
-        vnodeRegistrationTimeout = vnodeRegistrationTimeoutDefault
-        skipTestsDuringBuildCpis = this.skipTestsDuringBuildCpis
+        skipTestsDuringBuildCpis = "false"
+        cordaRuntimePluginWorkspaceDir = "workspace"
+        cordaBinDir = "${System.getProperty("user.home")}/.corda/corda5"
+        cordaCliBinDir = "${System.getProperty("user.home")}/.corda/cli"
+
+        // Alternatively, you can use a Database-only combined worker:
+        // composeFilePath = "config/combined-worker-compose.yaml"
```
Comment (anton-subbotin) [inline build.gradle]: We have an example of a Database-only compose file in the plugin's readme; the link to the readme was added to this repo's README. 

The question is - do we still want to provide a no-kafka example here in the template repo, including this comment and the compose yaml file in the config?

File: build.gradle [hunk 7]
```diff
@@ -110,7 +103,7 @@ allprojects {
 publishing {
     publications {
             maven(MavenPublication) {
-                artifactId "corda-CSDE-kotlin-sample"
+                artifactId "corda-dev-template-kotlin-sample"
```
Comment (anton-subbotin) [inline build.gradle:103]: I'm not sure if this is being used anywhere; publishing is disabled in the pipeline

File: build.gradle (hunk 6 above)
Comment (tlawson3) [reply build.gradle]: We should steer people to the Kafka enabled version, as it's quicker. We then offer the db only version to those who need it.

File: contracts/src/main/kotlin/com/r3/developers/cordapptemplate/utxoexample/contracts/ChatContract.kt (hunk 5 above)
Comment (tlawson3) [reply contracts/src/main/kotlin/com/r3/developers/cordapptemplate/utxoexample/contracts/ChatContract.kt:1]: Works for me

**Review by tlawson3 — COMMENTED** [review]
LGTM

File: build.gradle (hunk 6 above)
Comment (davidcurrie) [reply build.gradle]: I think it's fine to just have the Kafka version here. The non-Kafka version remains documented with the Gradle plugin.

File: build.gradle (hunk 7 above)
Comment (davidcurrie) [reply build.gradle:103]: I can't really see why it would ever be published.

File: config/combined-worker-compose.yaml [hunk 8]
```diff
@@ -0,0 +1,32 @@
+version: '2'
```
Comment (davidcurrie) [inline config/combined-worker-compose.yaml:1]: Let's just have the Kafka version...

File: config/combined-worker-kafka-compose.yaml [hunk 9]
```diff
@@ -0,0 +1,72 @@
+version: '2'
```
Comment (davidcurrie) [inline config/combined-worker-kafka-compose.yaml]: I've got some ideas for improvements we can make with this file but they can wait for another day.

File: contracts/src/main/kotlin/com/r3/developers/cordapptemplate/utxoexample/contracts/ChatContract.kt (hunk 5 above)
Comment (davidcurrie) [reply contracts/src/main/kotlin/com/r3/developers/cordapptemplate/utxoexample/contracts/ChatContract.kt:1]: Ditto

**Pull request closed without merging**